src/common/snippets/tests/src/lir_comparator.cpp

```diff
@@ -0,0 +1,234 @@
+            return Result::error(prefix + "Different ptr_increment: " + std::to_string(port.ptr_increment) + " and " +
+                                 std::to_string(port_ref.ptr_increment));
+        if (port.finalization_offset != port_ref.finalization_offset)
+            return Result::error(prefix + "Different finalization_offset: " + std::to_string(port.finalization_offset) + " and " +
+                                 std::to_string(port_ref.finalization_offset));
+        if (port.data_size != port_ref.data_size)
+            return Result::error(prefix + "Different data_size: " + std::to_string(port.data_size) + " and " +
+                                 std::to_string(port_ref.data_size));
+        if (port.dim_idx != port_ref.dim_idx)
+            return Result::error(prefix + "Different dim_idx: " + std::to_string(port.dim_idx) + " and " +
+                                 std::to_string(port_ref.dim_idx));
+
+        const auto& expr_port = port.expr_port;
+        const auto& expr_port_ref = port_ref.expr_port;
+        if (expr_port->get_type() != expr_port_ref->get_type())
+            return Result::error(prefix + "Different port types: " + std::to_string(expr_port->get_type()) + " and " +
+                                 std::to_string(expr_port_ref->get_type()));
+        if (expr_port->get_index() != expr_port_ref->get_index())
+            return Result::error(prefix + "Different port index: " + std::to_string(expr_port->get_index()) + " and " +
+                                 std::to_string(expr_port_ref->get_index()));
+        const auto& port_node = expr_port->get_expr()->get_node();
+        const auto& port_node_ref = expr_port_ref->get_expr()->get_node();
+        if (port_node->get_type_name() != port_node_ref->get_type_name())
+            return Result::error(prefix + "Different port expression type: " + port_node->get_type_name() + " and " +
+                                 port_node_ref->get_type_name());
+    }
+    return Result::ok();
+}
+
+LIRComparator::Result LIRComparator::compare_handlers(const SpecificIterationHandlers& handlers,
+                                                      const SpecificIterationHandlers& handlers_ref) {
+    auto compare_pipelines = [](const ov::snippets::lowered::pass::PassPipeline& pipeline,
+                                const ov::snippets::lowered::pass::PassPipeline& pipeline_ref) {
+        const auto& passes = pipeline.get_passes();
+        const auto& passes_ref = pipeline_ref.get_passes();
+        if (passes.size() != passes_ref.size())
+            return Result::error("Different passes count: " + std::to_string(passes.size()) + " and " +
+                                 std::to_string(passes_ref.size()));
+        for (size_t i = 0; i < passes.size(); ++i) {
+            const auto& pass = passes[i];
+            const auto& pass_ref = passes_ref[i];
+            if (pass->merge(pass_ref) == nullptr) {
```

Comment:
I used `merge` instead of RTTI comparison, because it is more strict check for now: it compares not only RTTI, but also fields of the passes.
>  Can two passes have different types, but still be mergeable (maybe not now, but in general).

But this point is absolutely valid, so it seems like `merge` can't be used in this case. Let's use RTTI comparison for now, but we will probably need to introduce `operator==` or some similar mechanism to perform more strict checks.

Reply:
Discussed offline: now both checks are performed